gulpfile.js

```diff
@@ -86,28 +99,104 @@ gulp.task('test', function() {
+buildTask('stats', [JSONReader(options.file), treeBuilderWriter(StatsWriter), consoleOutput()]);
+buildTask('compactComputedStyle', [JSONReader(options.file), filter(StyleFilter), fileOutput(options.file + '.filter')]);
+buildTask('extractStyle', [JSONReader(options.file), filter(StyleMinimizationFilter), fileOutput(options.file + '.filter')]);
+buildTask('generate', [JSONReader(options.file), fabricator(SchemaBasedFabricator), fileOutput(options.file + '.gen')]);
+buildTask('tokenStyles', [JSONReader(options.file), filter(StyleTokenizerFilter), fileOutput(options.file + '.filter')]);
+buildTask('nukeIFrame', [JSONReader(options.file), filter(NukeIFrameFilter), fileOutput(options.file + '.filter')]);
+buildTask('runExperiment', [fileReader(options.file), parseExperiment(), runExperiment, consoleOutput()]);
+
+/*
+ * experiments
+ */
+
+function collectInputs(inputSpec) {
+  var files = fs.readdirSync('.');
+  return files.filter(inputSpec.exec.bind(inputSpec));
+}
+
+function outputForInput(inputSpec, input, output) {
+  var re = new RegExp(inputSpec);
+  return input.replace(re, output);
+}
+
+function appendEdges(experiment, stages, edges) {
+  var newList = [];
+  for (var j = 0; j < edges.length; j++) {
+    var newStages = stages.concat(edges[j].stages);
+    if (edges[j].output in experiment.tree)
+      newList = newList.concat(appendEdges(experiment, newStages, experiment.tree[edges[j].output]));
+    else
+      newList.push({stages: newStages, output: edges[j].output});
+  }
+  return newList;
+}
+
+function experimentTask(name, experiment) {
+  gulp.task(name, function(cb) { runExperiment(experiment, cb); });
+}
+
+function runExperiment(experiment, cb) {
+  var pipelines = [];
+  for (var i = 0; i < experiment.inputs.length; i++) {
+    var inputs = collectInputs(new RegExp('^' + experiment.inputs[i] + '$'));
+    var edges = experiment.tree[experiment.inputs[i]];
+    var stagesList = [];
+    stagesList = appendEdges(experiment, stagesList, edges);
+    for (var j = 0; j < inputs.length; j++) {
+      for (var k = 0; k < stagesList.length; k++) {
+	// FIXME: This relies on the fact that filters and writers are both the same thing
+	// right now (i.e. filter and treeBuilderWriter are the same function).
+        // This could well become a problem in the future.
+        // Also, eval: ew. If there was a local var dict I could look up the constructor name directly.
+        var pl = [JSONReader(inputs[j])].concat(stagesList[k].stages.map(function(a) { return filter(eval(a)); }));
+        pl.push(fileOutput(outputForInput(experiment.inputs[i], inputs[j], stagesList[k].output)));
+        pipelines.push(pl);
+      }
+    }
+  }
+  for (var i = 0; i < pipelines.length; i++) {
+    var cb = (function(i, cb) { return function() { processStages(pipelines[i], cb); } })(i, cb);
+  }
+  return cb(null);
+}
+
+experimentTask('experiment', 
```

Comment:
Is it worth keeping this? It's an example of how to manually set up a gulp build target using the experiment data structures.